app/shop.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from datetime import datetime
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list
+    products: dict
+
+    def products_cost(self, cart: dict) -> dict:
+        return {
+            product: cart[product] * self.products[product]
+            for product in cart
+        }
+
+    def shopping(self, customer_name: str, cart: dict) -> None:
+        print("Date: ", datetime.now().strftime("%d/%m/%Y %H:%M:%S"))
+        print(f"Thanks, {customer_name}, for your purchase!")
+        print("You have bought:")
+
+        products_cost = self.products_cost(cart)
+        for product in products_cost:
+            print(f"{cart[product]} {product}s for "
+                  f"{self.round_price(products_cost[product])} dollars")
+
+        print(f"Total cost is {sum(self.products_cost(cart).values())}"
+              f" dollars")
+        print("See you again!\n")
```

Comment:
The `round_price` method is designed to return an integer if the price is a whole number. Ensure that this behavior is intended, as it might affect how prices are displayed in the shopping receipt.